File: Day 32/cuadraticEq.js
console.log("The equation I'll solve is: a·x^2 + b·x + c = 0")

let a = prompt("Type the first coeficient (a):")
let b = prompt("Type the second coeficient (b):")
let c = prompt("Type the independent coeficient (c):")

function solveCuadEq(a, b, c){
    
    let r = Math.pow( b, 2 ) - 4*a*c

    if( r < 0 )
        console.log(`The equation ${a}·x^2 + ${b}·x + ${c} = 0 hasn't solution in the real numbers`)
    else if(a === 0){
        let x = -c/b
        console.log(`The solution for ${b}·x + ${c} = 0 is x=${x}`)
    }
    else if(r === 0){
        let x = -b/(2*a)
        console.log(`The solution for ${a}·x^2 + ${b}·x + ${c} = 0 is double x=${x}`)
    }
    else{
        let x1 = ( -b + Math.sqrt(r) )/(2*a)
        let x2 = ( -b - Math.sqrt(r) )/(2*a)
        console.log(`The solutions for ${a}·x^2 + ${b}·x + ${c} = 0 are x1=${x1} and x2=${x2}`)
    }

}

solveCuadEq(a, b, c)
solveCuadEq(0, b, c)
solveCuadEq(a, 0, c)
solveCuadEq(a, b, 0)

File: Day 32/cuadraticEq.c
#include <stdio.h>
#include <math.h>

void solveEq(int a, int b, int c);

int main()
{

    int a, b, c;
    printf("Type the coeficients a, b and c: \n");
    scanf("%d", &a);
    scanf("%d", &b);
    scanf("%d", &c);

    solveEq(a, b, c);

    return 0;
}

void solveEq(int a, int b, int c){

    float r = pow( b, 2 ) - 4*a*c;

    if( r < 0 )
        printf("The equation %dx^2 + %dx + %d = 0 has not solution in the real numbers", a, b, c);
    else if(a == 0){
        float x = -c/b;
        printf("The solution for %dx + %d = 0 is x=%f", b, c, x);
    }
    else if(r == 0){
        float x = -b/(2*a);
        printf("The solution for %dx^2 + %dx + %d = 0 is double x=%f", a, b, c, x);
    }
    else{
        float x1 = ( -b + sqrt(r) )/(2*a);
        float x2 = ( -b - sqrt(r) )/(2*a);
        printf("The solutions for %dx^2 + %dx + %d = 0 are x1=%f and x2=%f", a, b, c, x1, x2);
    }

}